### ISSUE TITLE
Allow wildcards for Hive ANALYZE

### ISSUE THREAD

**Issue by electrum** [prestosql/presto#962]
It seems useful to allow analyzing multiple partitions without explicitly listing them all (and without analyzing the entire table). We need to figure out a good way to represent this using the property syntax.

**Comment by findepi** [comment]
Another idea would be to allow `ANALYZE t WHERE ..` (just the same way we do with `DELETE`).
This would have additional advantage that the syntax would be much easier. `WITH(partitions = ARRAY[ARRAY['2018-01-01']])` is not as easy to write as `WHERE dt = DATE '2018-01-01'`.

**Comment by martint** [comment]
@findepi, that was one of the approaches we considered originally, but there were some issues with it. I’ll try to dig up the discussions.

**Comment by findepi** [comment]
@martint yeah, i was under impression that it has been considered.... So I'm challenging that decision 😉

**Comment by sopel39** [comment]
We could potentially have two approaches in parallel, one for simpler cases `ANALYZE t WHERE ...` and one for more advanced cases.

**Comment by tnatssb** [comment]
Bringing this back to life. As the CBO and the stats become vitally important, more education around collecting them is becoming apparent. 

The additional features would be nice here would be:
1. Collect all sub-partitions using the analyze command. Example: month= and day=. Analyze on month would include all of the day= partitions
2. Analyze with wildcards/regex. Example: table is partitioned by day_id=YYYYMMDD so you can anayze by day_id=202101[1-9][1-9] something like that where you could analyze all the days in the month.

@findepi
